### ISSUE TITLE
Jupyter finds kernels that shouldn't exist.

### ISSUE THREAD

**Issue by jpmvferreira** [jupyterhub/jupyterhub#2052]
Hi there,

I've recently starting using Jupyter and Anaconda, and after trying to setup everything with my default kernel (python 3) and another kernel (my environment called "mobilidade" with python 2.7) i open up Jupyter to find this:

![image](https://user-images.githubusercontent.com/41581666/43081196-64386c54-8e89-11e8-8f3c-7731701ab6cf.png)

And when i check for my environments in Jupyter i get:

![image](https://user-images.githubusercontent.com/41581666/43081252-82415f9e-8e89-11e8-83f2-92f5a0a93c47.png)

I don't know where i got this anaconda3 environment, since it doesn't even exist (error message given by Jupyter itself).
To troubleshoot it I've tried to create and then delete a environment called anaconda3 but it didn't work.
Also there are two default kernels when i start Jupyter from my base environment:

![image](https://user-images.githubusercontent.com/41581666/43081356-b860b6d8-8e89-11e8-9df8-077d4be0bced.png)

Whenever i try to delete anaconda3 , the weird spooky environment, i get this message:

![image](https://user-images.githubusercontent.com/41581666/43081987-d4268266-8e8a-11e8-9bd6-0f11890c3431.png)

When i use the command `jupyter kernelspec list` on my root and custom environment in the anaconda prompt i get:

![image](https://user-images.githubusercontent.com/41581666/43081748-3fa2a6ec-8e8a-11e8-91ea-933485db4314.png)

I have seen this thread https://github.com/altair-viz/altair/issues/837 to try and fix this issue but it didn't work, since even if i type `jupyter notebook` after activating my environment, i still get all of those redundant environments in Jupyter, along with the non-existing anaconda3 environment.

This isn't game breaking or anything, it's just a matter of keeping things tidy and organized.

Thanks in advance!

**TL;DR**: _nb_conda_kernels finds 4 kernels, however i only have 1 environment and the root location to work with._

**Comment by willingc** [comment]
I believe this had been an nbconda bug. Please reference the discussion [here](https://stackoverflow.com/questions/48012811/jupyter-erroneous-environments-showing-up-in-conda-tab-with-nb-conda-extension)
